awx/ui_next/src/util/qs.test.js

```diff
@@ -540,12 +511,237 @@ describe('qs (qs.js)', () => {
+      expect(_stringToObject(config, '?unit.foo=bar&unit.foo=bam')).toEqual({
+        foo: ['bar', 'bam'],
+      });
+    });
+
+    test('should omit keys from other namespaces', () => {
+      const config = { namespace: 'unit' };
+      expect(
+        _stringToObject(config, '?unit.foo=bar&other.bar=bam&one=two')
+      ).toEqual({
+        foo: 'bar',
+      });
+    });
+
+    test('should convert numbers to correct type', () => {
+      const config = {
+        namespace: 'unit',
+        integerFields: ['page'],
+      };
+      expect(_stringToObject(config, '?unit.page=3')).toEqual({
+        page: 3,
+      });
+    });
+  });
+
+  describe('_addDefaultsToObject', () => {
+    test('should add missing default values', () => {
+      const config = {
+        defaultParams: { page: 1, page_size: 5, order_by: 'name' },
+      };
+      expect(_addDefaultsToObject(config, {})).toEqual({
+        page: 1,
+        page_size: 5,
+        order_by: 'name',
+      });
+    });
+
+    test('should not override existing params', () => {
+      const config = {
+        defaultParams: { page: 1, page_size: 5, order_by: 'name' },
+      };
+      const params = {
+        page: 2,
+        order_by: 'date_created',
+      };
+      expect(_addDefaultsToObject(config, params)).toEqual({
+        page: 2,
+        page_size: 5,
+        order_by: 'date_created',
+      });
+    });
+
+    test('should handle missing defaultParams', () => {
+      const params = {
+        page: 2,
+        order_by: 'date_created',
+      };
+      expect(_addDefaultsToObject({}, params)).toEqual({
+        page: 2,
+        order_by: 'date_created',
+      });
+    });
+  });
+
+  describe('mergeParams', () => {
+    it('should merge param into an array', () => {
+      const oldParams = {
+        foo: 'one',
+      };
+      const newParams = {
+        foo: 'two',
+      };
+      expect(mergeParams(oldParams, newParams)).toEqual({
+        foo: ['one', 'two'],
+      });
+    });
+
+    it('should retain unaltered params', () => {
+      const oldParams = {
+        foo: 'one',
+        bar: 'baz',
+      };
+      const newParams = {
+        foo: 'two',
+      };
+      expect(mergeParams(oldParams, newParams)).toEqual({
+        foo: ['one', 'two'],
+        bar: 'baz',
+      });
+    });
+
+    it('should gather params from both objects', () => {
+      const oldParams = {
+        one: 'one',
+      };
+      const newParams = {
+        two: 'two',
+      };
+      expect(mergeParams(oldParams, newParams)).toEqual({
+        one: 'one',
+        two: 'two',
+      });
+    });
+
+    it('should append to array', () => {
+      const oldParams = {
+        foo: ['one', 'two'],
+      };
+      const newParams = {
+        foo: 'three',
+      };
+      expect(mergeParams(oldParams, newParams)).toEqual({
+        foo: ['one', 'two', 'three'],
+      });
+    });
+
+    it('should append array to existing value', () => {
```

Comment:
Is this test needed?  It seems to be testing the same thing as the test above.

Reply:
they are three slightly different cases: val + val => array, array + val => array, val + array => array, each of which exercise different paths of the code